client/state/posts/reducer.js

```diff
@@ -163,40 +170,33 @@ export const queries = ( () => {
 
 	return createReducer( {}, {
 		[ POST_RESTORE ]: ( state, { siteId, postId } ) => {
-			return applyToManager( state, siteId, 'receive', {
+			return applyToManager( state, siteId, 'receive', false, {
 				ID: postId,
 				status: '__RESTORE_PENDING'
 			}, { patch: true } );
 		},
 		[ POST_RESTORE_FAILURE ]: ( state, { siteId, postId } ) => {
-			return applyToManager( state, siteId, 'receive', {
+			return applyToManager( state, siteId, 'receive', false, {
 				ID: postId,
 				status: 'trash'
 			}, { patch: true } );
 		},
 		[ POSTS_REQUEST_SUCCESS ]: ( state, { siteId, query, posts, found } ) => {
-			if ( ! state[ siteId ] ) {
-				state = {
-					...state,
-					[ siteId ]: new PostQueryManager()
-				};
-			}
-
-			return applyToManager( state, siteId, 'receive', posts, { query, found } );
+			return applyToManager( state, siteId, 'receive', true, posts, { query, found } );
 		},
 		[ POSTS_RECEIVE ]: ( state, { posts } ) => {
 			return reduce( groupBy( posts, 'site_ID' ), ( memo, sitePosts, siteId ) => {
-				return applyToManager( memo, siteId, 'receive', sitePosts );
+				return applyToManager( memo, siteId, 'receive', true, sitePosts );
```

Comment:
>Change in behavior here: this will create a new `PostQueryManager` if one does not exist.  When would this happen?  Does the new behavior need a test?

Correct. Since previously queries was only responsible for tracking posts that were initiated by a specific request query, we'd not wanted to create a new instance for a site if it's not already tracked (that was previously the responsibility of `items`). Now that we're consolidating the reducers, we _do_ want to track any ad-hoc posts that are received, regardless of whether they're associated with a particular query. I can't think of a specific case in which this would occur currently, but an example future situation might be: receiving the post to be edited in the post editor, reader posts.

Adding a test for this case would probably be a good idea, yes.